tests/integration/dhcp_test.py

```diff
@@ -0,0 +1,154 @@
+#
+# Copyright 2018 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+import os
+import pytest
+
+from libnmstate import netapplier
+from libnmstate import nm
+
+from .testlib import assertlib
+from .testlib import statelib
+from .testlib.statelib import INTERFACES
+from .testlib import cmd as libcmd
+
+
+IPV4_ADDRESS1 = '192.0.2.251'
+IPV4_ADDRESS2 = '192.0.2.252'
+IPV6_ADDRESS1 = '2001:db8:1::1'
+IPV6_ADDRESS2 = '2001:db8:2::1'
+
+DHCP_SRV_NIC = 'dhcpsrv'
+DHCP_CLI_NIC = 'dhcpcli'
+
+DNSMASQ_CONF_STR = """
+interface={}
+dhcp-range=192.0.2.200,192.0.2.250,255.255.255.0,48h
+enable-ra
+dhcp-range=2001:db8:1::100,2001:db8:1::fff,64,480h
+""".format(DHCP_SRV_NIC)
+
+DNSMASQ_CONF_PATH = '/etc/dnsmasq.d/nmstate.conf'
+
+SYSFS_DISABLE_IPV6_FILE = '/proc/sys/net/ipv6/conf/{}/disable_ipv6'.format(
+    DHCP_SRV_NIC)
+
+# Python 2 does not have FileNotFoundError and treat file not exist as IOError
+try:
+    FileNotFoundError
+except NameError:
+    FileNotFoundError = IOError
+
+
+@pytest.fixture(scope='module')
+def start_dhcpd(request):
```

Comment:
Please wrap the commands using functions which describe what the commands do:

- create_veth (probably setting its link up as well)
- Set addresses.
- Bind dnsmasq on a nic.
etc...

It should read nicely leaving the details deep in.